### PR TITLE
Refactored logger logic

### LINKED ISSUE #1106: Refactor logger to have a separated transport module
Transports.js should generate all transport objects for the logger rather than this being baked into the logger itself

### PR DESCRIPTION
**Pull request by chrismaree**
This PR will close #1106. This PR makes the logger more modular, de-coupling elements between transports, winston and how each section is initialized. 

All logger logic has been moved into the `logger` directory. `Logger.js` imports winston and `Transport.js` and creates the logger instances. `Transport.js` builds up an array of transports based on the present environment variables. For now this consists of three distinct transports:

1. a console transport that logs at all levels. This is imported from `ConsoleTransport` which defines a formatter and console transporter.
2. a slack transport that logs level `info` and `error`. it requires a `SLACK_WEBHOOK` env variable to operate. It is initialized by `SlackTransport.js` which leverages `winston-slack-webhook-transport`.
3. a twilio transporter that logs at level `error`. it requires a minimum of env variables: `TWILIO_SID && TWILIO_AUTH && DRI_NUMBER1 && TWILIO_FROM_NUMBER` to operate. description of these is shown in the `env_sample`. this transport imports from RoboCaller to place calls.

All logic to create transports, parse env variables and sanitize inputs is placed within `Transport.js` to achieve separation of concerns.

### CODE REVIEW

File: financial-templates-lib/logger/ConsoleTransport.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// This transport enables Winston logging to the console.
+const winston = require("winston");
+
+const alignedWithColorsAndTime = winston.format.combine(
+  winston.format.colorize(),
+  winston.format.timestamp(),
+  winston.format.printf(info => {
+    const { timestamp, level, ...args } = info;
+
+    const ts = timestamp.slice(0, 19).replace("T", " ");
```
Comment (nicholaspai) [inline financial-templates-lib/logger/ConsoleTransport.js]: What's the significance of the `19`?

File: financial-templates-lib/logger/Logger.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// The logger has a number of different levels based on the severity of the incident:
+//-> Debugs: self explanatory. Normal status-based logging. These can trigger
+//   every iteration. Unlimited volume.
+//-> Info: when something happens that is notable, but not necessarily actionable.
+//   These should not trigger every iteration. Any on-chain event that executed correctly.
+//   these trigger a slack message that everyone has access to.
+//-> Error: anything that requires human intervention. If the bot is low on funds or a
```
Comment (nicholaspai) [inline financial-templates-lib/logger/Logger.js]: Question for you: Does logger.error also print out errors? I've noticed when running the `liquidator/test/liquidator.js` tests, the last test correctly catches an error but also prints out the error which may not be as intended?

File: financial-templates-lib/logger/SlackTransport.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+// This transport enables slack messages to be sent from twillio logging. To configure this
+// create a slack webhook and add this to your .env file. a sample in .env_sample shows this.
+// see https://slack.com/intl/en-za/help/articles/115005265063-Incoming-Webhooks-for-Slack for more.
+
+const SlackHook = require("winston-slack-webhook-transport");
+
+const slackFormatter = info => {
+  if (!("level" in info) || !("at" in info) || !("message" in info)) {
+    console.error("WINSTON INCORRECTLY CONFIGURED IN MESSAGE", info);
```
Comment (nicholaspai) [inline financial-templates-lib/logger/SlackTransport.js]: I like this check.

File: financial-templates-lib/logger/Transport.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+if (process.env.SLACK_WEBHOOK) {
+  transports.push(SlackTransport.createSlackTransport(process.env.SLACK_WEBHOOK));
+}
+
+// If all the required environment variables for twilio are added, add the transport array.
+if (process.env.TWILIO_SID && process.env.TWILIO_AUTH && process.env.DRI_NUMBER1 && process.env.TWILIO_FROM_NUMBER) {
+  // read in the numbers to call from the environment variables. Each number is prefaced by a DRI_NUMBER
+  let numbersToCall = [];
+  for (const envVariable in process.env) {
+    if (envVariable.startsWith("DRI_NUMBER")) {
```
Comment (nicholaspai) [inline financial-templates-lib/logger/Transport.js]: I think we should make the `.env.sample` have comments explaining what each environment variable does. And how you can have multiple DRI_NUMBER_X's.

https://stackoverflow.com/questions/26713508/is-there-a-comment-character-for-foremans-env-file

File: financial-templates-lib/logger/ConsoleTransport.js (hunk 1 above)
Comment (chrismaree) [reply financial-templates-lib/logger/ConsoleTransport.js]: it formats the time stamp to look like this `2020-03-24 13:27:08`

File: financial-templates-lib/logger/Logger.js (hunk 2 above)
Comment (chrismaree) [reply financial-templates-lib/logger/Logger.js]: good q. I will investigate this and get back to you. might be a seperate error.

File: financial-templates-lib/logger/Transport.js (hunk 4 above)
Comment (chrismaree) [reply financial-templates-lib/logger/Transport.js]: There are detailed comments already from the previous commit in the [sample](https://github.com/UMAprotocol/protocol/blob/master/core/.env_sample). Please let me know if you need more.

File: financial-templates-lib/logger/ConsoleTransport.js (hunk 1 above)
Comment (nicholaspai) [reply financial-templates-lib/logger/ConsoleTransport.js]: ok, just add a comment showing example output

File: financial-templates-lib/logger/Transport.js (hunk 4 above)
Comment (nicholaspai) [reply financial-templates-lib/logger/Transport.js]: all good!

File: financial-templates-lib/logger/Logger.js (hunk 2 above)
Comment (chrismaree) [reply financial-templates-lib/logger/Logger.js]: this is expected behaviour. I updated the comments accordingly.

**Pull request merged**